src/core/build_label.go

```diff
@@ -137,67 +136,27 @@ func ParseBuildLabel(target, currentPath string) BuildLabel {
 	return label
 }
 
-func splitAnnotation(target string) (string, string) {
+// SplitLabelAnnotation splits the build label from the annotation
+func SplitLabelAnnotation(target string) (string, string) {
```

Comment:
Still not entirely happy with this. Ideally consumers of this package wouldn't have to worry about annotations. Everything depends on the concrete `BuildLabel` type so we can't add indirection without a huge refactor, and adding the annotation to that struct increases the memory footprint quite a bit because we construct potentially 100k or even millions of these during a build.